pkg/machinery/marker.go

```diff
@@ -25,9 +25,11 @@ import (
 const prefix = "+kubebuilder:scaffold:"
 
 var commentsByExt = map[string]string{
-	".go":   "//",
-	".yaml": "#",
-	".yml":  "#",
+	".go":        "//",
+	".work":      "//",
+	".yaml":      "#",
+	".yml":       "#",
+	"Dockerfile": "#",
```

Comment:
Why we need to add the Dockerfile here?

Reply:
As you can see in the initial comment of the PR, the Dockerfile has to be adjusted to stay optimized, otherwise the go mod download command will only download dependencies of the main module, not of API modules as well. If we can live with that, we do not have to touch the Dockerfile, but I would still say that we should optimize it here.

Reply:
HI @jakobmoellersap, 

We can customize the Dockerfile, as we do for example with the declarative plugin. See that you can use the plugin utils to insert, replace code.

This implementation is for we create markers to inject code. Are we creating any marker for the dockerfile or these other extensions?

Reply:
this would have required a marker right now. But I think we can also use the way that the declarative plugin is using